packages/backend/src/modules/tvl2/indexers/PriceIndexer.ts

```diff
@@ -37,97 +35,56 @@ export class PriceIndexer extends ManagedMultiIndexer<CoingeckoPriceConfigEntry>
     to: number,
     configurations: UpdateConfiguration<CoingeckoPriceConfigEntry>[],
   ): Promise<number> {
-    const configurationsWithMissingData = configurations.filter(
-      (c) => !c.hasData,
-    )
-    if (configurationsWithMissingData.length === 0) {
-      return to
+    const configurationsToSync = configurations.filter((c) => !c.hasData)
+
+    if (configurationsToSync.length !== configurations.length) {
+      this.logger.info('Filtered out configurations with data', {
+        skippedConfigurations:
+          configurations.length - configurationsToSync.length,
+        configurationsToSync: configurationsToSync.length,
+        from,
+        to,
+      })
     }
 
-    const prices = await this.fetchAndOptimizePrices(
-      from,
-      to,
-      configurationsWithMissingData,
-    )
-
-    if (prices.length === 0) {
+    if (configurationsToSync.length === 0) {
+      this.logger.info('No configurations to sync', {
+        from,
+        to,
+      })
       return to
     }
 
-    // There are multiple price configs with possibly different until timestamps.
-    // We need to find the latest until timestamp to return.
-    const maxTo = prices
-      .map((p) => p.timestamp.toNumber())
-      .reduce((a, b) => Math.max(a, b), 0)
-
-    this.logger.info('Fetched & optimized prices in range', {
-      from,
-      to: maxTo,
-      prices: prices.length,
-      configurations: configurationsWithMissingData.length,
-    })
+    const adjustedTo = this.$.priceService.getAdjustedTo(from, to)
```

Comment:
Nit pick: this function could be called `getAdjustedRange` and return `{from: UnixTime, to: UnixTime}`